File: Source/Service/appService/com.maxiaohua.genealogy.main.biz/src/com/maxiaohua/genealogy/main/biz/a01/dto/FamilyInfoDTO.java
/*
 * Genealogy
 * Xiao Hua Ma personal studio 2015.
 */
package com.maxiaohua.genealogy.main.biz.a01.dto;

/**
 * FamilyInfoDTO DTO
 *
 * History
 * REV.         Updated Date           Updater              Infomation
 * -------      ---------------        ----------------     ------------------
 * 1.0          2015/09/06             TOOL                Create
*
 */
@com.maxiaohua.genealogy.fw.core.type.AliasKanJi("家族信息")
public class FamilyInfoDTO implements java.io.Serializable, Comparable<FamilyInfoDTO> {
	/**
	* System ID
	*/
	private static final long serialVersionUID = 1L;

	private String familyID = null;

	public String getFamilyID() {
		return this.familyID;
	}

	public void setFamilyID(
			String familyID) {
		this.familyID = familyID;
	}

	private String arms = null;

	public String getArms() {
		return this.arms;
	}

	public void setArms(
			String arms) {
		this.arms = arms;
	}

	private String famileyName = null;

	public String getFamileyName() {
		return this.famileyName;
	}

	public void setFamileyName(
			String famileyName) {
		this.famileyName = famileyName;
	}

	private String introductionText = null;

	public String getIntroductionText() {
		return this.introductionText;
	}

	public void setIntroductionText(
			String introductionText) {
		this.introductionText = introductionText;
	}

	private String introductionPicture = null;

	public String getIntroductionPicture() {
		return this.introductionPicture;
	}

	public void setIntroductionPicture(
			String introductionPicture) {
		this.introductionPicture = introductionPicture;
	}

	private String introductionVideo = null;

	public String getIntroductionVideo() {
		return this.introductionVideo;
	}

	public void setIntroductionVideo(
			String introductionVideo) {
		this.introductionVideo = introductionVideo;
	}

  	/* (non-Javadoc)
     * @see java.lang.Comparable#compareTo(java.lang.Object)
     */
    @Override
    public int compareTo(
    		FamilyInfoDTO compareToDTO) {
        return this.equals(compareToDTO) ? 0 : -1;
    }
}

File: Source/iOS/Genealogy/Genealogy/Model/A01/DTO/FamilyInfoDTO.m
//
//  FamilyInfoDTO.m
//  Genealogy
//
//  Created by xiao huama on 15/09/10.
//  Xiao Hua Ma personal studio 2015.
//

#import "FamilyInfoDTO.h"

@implementation FamilyInfoDTO

- (void)loadDataWithJsonData:(id)jsonData
{
    if (![jsonData isKindOfClass:[NSDictionary class]]) {
        return;
    }
    if (!jsonData) {
        return;
    }
    if (![jsonData[@"familyID"] isKindOfClass:[NSNull class]]) {
        self.familyID = jsonData[@"familyID"];
    }
    if (![jsonData[@"arms"] isKindOfClass:[NSNull class]]) {
        self.arms = jsonData[@"arms"];
    }
    if (![jsonData[@"famileyName"] isKindOfClass:[NSNull class]]) {
        self.famileyName = jsonData[@"famileyName"];
    }
    if (![jsonData[@"introductionText"] isKindOfClass:[NSNull class]]) {
        self.introductionText = jsonData[@"introductionText"];
    }
    if (![jsonData[@"introductionPicture"] isKindOfClass:[NSNull class]]) {
        self.introductionPicture = jsonData[@"introductionPicture"];
    }
    if (![jsonData[@"introductionVideo"] isKindOfClass:[NSNull class]]) {
        self.introductionVideo = jsonData[@"introductionVideo"];
    }
}

-(NSMutableDictionary *)buildRequestData
{
    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
    if (self.familyID) {
        dic[@"familyID"] = self.familyID;
    }
    if (self.arms) {
        dic[@"arms"] = self.arms;
    }
    if (self.famileyName) {
        dic[@"famileyName"] = self.famileyName;
    }
    if (self.introductionText) {
        dic[@"introductionText"] = self.introductionText;
    }
    if (self.introductionPicture) {
        dic[@"introductionPicture"] = self.introductionPicture;
    }
    if (self.introductionVideo) {
        dic[@"introductionVideo"] = self.introductionVideo;
    }
    return dic;
}

@end
